### PR TITLE
[RFC] Add Xilinx Vivado Design Suite

### PR DESCRIPTION
**Pull request by corna**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ ] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *see below*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *No patches*

-----

Hi all.

As marked in the title, I'd like some suggestion regarding this package.

This PR adds the [Xilinx Vivado Design Suite](https://www.xilinx.com/downloads) as flatpak package. All the Xilinx material is NOT redistributable and requires a login to be downloaded (and the filling of a "U.S. Government Export Approval" form). For this reason, this repository does not contain any Xilinx code and instead works as a wrapper over an "official" installation. The user is prompted to download the official installation tool, which is then passed to this package and installed within the sandbox (in the .Xilinx folder).

### Permissions
 - `share=ipc`, `socket=wayland` and `socket=x11` are obvious.
 - `share=network` is required for the correct functionality of docnav (which downloads the documentation from the internet) and is required for licensing (as the license is bound to the MAC address). 
 - `device=all` is required for accessing the hardware devices (USB), which can be programmed through Vivado. For this, the user has to install some udev rules, see below.
 - `filesystem=~/.Xilinx:create`: this folder is used by the Xilinx tools to store the data and licenses. Additionally, it is the default install location used by the helper scripts.
 - `filesystem=xdg-documents`: the user has to store its project somewhere. I don't know however if it is better to provide just the documents (better sandbox) or the whole home directory (more user-friendly).

### xilinx.com

Regarding [the requirements][reqs], I have not contacted Xilinx regarding this package (I don't even know where I should contact them, there does not seem to be an email for non-paying customers). I also do not own the xilinx.com domain of course, so I don't know if this package should be left under this ID or not.

### Icons

The better-suited icons for this package (to be consistent with an official installation) would be the Vivado icon, the Vitis icon and the Xilinx logo. However, if I got it correctly, icons have to be packaged into the flatpak package at build time, which makes it hard since these icons are copyrighted (and trademarks I believe). Suggestions?

### udev rules

The tools can program the boards through many USB-JTAG adapters, which however require an udev rule to correctly set the permissions. Is there a way to handle this from flatpak? The easy solution is to tell the user to run a command as sudo, but it is not the best in terms of user-friendliness.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.xilinx.Vivado

**Comment by barthalion** [comment]
Appdata needs some fixes:

```
builddir/files/share/appdata/com.xilinx.Vivado.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• style-invalid         : <summary> is too short [TODO] minimum is 8 chars
• tag-missing           : <url> is not present
```

**Comment by corna** [comment]
bot, build com.xilinx.Vivado

**Comment by barthalion** [comment]
bot, build com.xilinx.Vivado

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["i386", "x86_64"]
```
Comment (bilelmoussaoui) [inline flathub.json:2]:
```suggestion
  "only-arches": ["x86_64"]
```

We don't support i386 architectures

File: resources/share/metainfo/com.xilinx.Vivado.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2021 Nicola Corna <nicola@corna.info> -->
+<application>
```
Comment (bilelmoussaoui) [inline resources/share/metainfo/com.xilinx.Vivado.appdata.xml:3]: That's the old appdata format, you should run `appstream-util upgrade` on it.

File: resources/share/metainfo/com.xilinx.Vivado.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  <id type="desktop">com.xilinx.Vivado.desktop</id>
```
Comment (bilelmoussaoui) [inline resources/share/metainfo/com.xilinx.Vivado.appdata.xml:4]:
```suggestion
  <id>com.xilinx.Vivado</id>
  <launchable type="desktop-id">com.xilinx.Vivado.desktop</launchable>
```

**Comment by corna** [comment]
Superseded by #2164

**Pull request closed without merging**